src/typy.js

```diff
@@ -165,6 +189,11 @@ class Typy {
     if (!this.isNullOrUndefined) return [this.input];
     return [];
   }
+
+  get safeDate() {
+    if (this.isDate) return this.input;
+    return null;
```

Comment:
@rehman-00001 Can we return `undefined` instead of `null` here and update the doc changes as well. We tend to return `undefined` whenever an intended obj/key is not found. We'd like to keep it that way for consistency.

As to why, `undefined` is essentially meaning to nothing in JavaScript. `null` is a value and can be used to compare against other values in JavaScript. It doesn't make sense for a library to return null as a value when the value essentially doesn't exist or can be found.

Reply:
@flexdinesh Hmm.. Actually I was in a dilemma, whether to include this getter or not. Just as the name indicates, `safeDate` should return a "safe" date value. In case of other similar getters (`safeBoolean`, `safeObject`, `safeArray`, etc) the default value specific to its type is returned. 

Those getters are reliable and can be used in situations like these:

```
t(someObject, 'someKey.myObj').safeObject.newProperty = 'someValue';
t(someObject, 'someKey.myArray').safeArray.push({ someObj });
```

But in case of `safeDate` since there's no default Date value, it either have to return `undefined` or `null`. And I felt that might violate the meaning of "safe" access. So I left this decision to the creator @flexdinesh. :thinking:   Is it okay to have this getter `safeDate` returning `undefined` or can we just remove it?